frontend/src/components/widgets/DataFrame/hooks/useDataLoader.ts

```diff
@@ -0,0 +1,318 @@
+export const INDEX_IDENTIFIER = "index"
+// Prefix used in the config column mapping when referring to a column via the numeric position
+export const COLUMN_POSITION_PREFIX = "col:"
+
+/**
+ * Options to configure columns.
+ */
+export interface ColumnConfigProps {
+  width?: number
+  title?: string
+  type?: string
+  hidden?: boolean
+  editable?: boolean
+  metadata?: Record<string, unknown>
+  alignment?: string
+}
+
+/**
+ * Apply the user-defined column configuration if supplied.
+ *
+ * @param columnProps - The column properties to apply the config to.
+ * @param columnConfigMapping - The user-defined column configuration mapping.
+ *
+ * @return the column properties with the config applied.
+ */
+export function applyColumnConfig(
+  columnProps: BaseColumnProps,
+  columnConfigMapping: Map<string | number, ColumnConfigProps>
+): BaseColumnProps {
+  if (!columnConfigMapping) {
+    // No column config configured
+    return columnProps
+  }
+
+  let columnConfig
+  if (columnConfigMapping.has(columnProps.title)) {
+    // Config is configured based on the column title
+    columnConfig = columnConfigMapping.get(columnProps.title)
+  } else if (
+    columnConfigMapping.has(
+      `${COLUMN_POSITION_PREFIX}${columnProps.indexNumber}`
+    )
+  ) {
+    // Config is configured based on the column position, e.g. col:0 -> first column
+    columnConfig = columnConfigMapping.get(
+      `${COLUMN_POSITION_PREFIX}${columnProps.indexNumber}`
+    )
+  } else if (
+    columnProps.isIndex &&
+    columnConfigMapping.has(INDEX_IDENTIFIER)
+  ) {
+    // Config is configured for the index column (or all index columns for multi-index)
+    columnConfig = columnConfigMapping.get(INDEX_IDENTIFIER)
+  }
+
+  if (!columnConfig) {
+    // No column config found for this column
+    return columnProps
+  }
+
+  return {
+    ...columnProps,
+    // Update title:
+    ...(notNullOrUndefined(columnConfig.title)
+      ? {
+          title: columnConfig.title,
+        }
+      : {}),
+    // Update width:
+    ...(notNullOrUndefined(columnConfig.width)
+      ? {
+          width: columnConfig.width,
+        }
+      : {}),
+    // Select a column type:
+    ...(notNullOrUndefined(columnConfig.type)
+      ? {
+          customType: columnConfig.type.toLowerCase().trim(),
+        }
+      : {}),
+    // Update editable state:
+    ...(notNullOrUndefined(columnConfig.editable)
+      ? {
+          isEditable: columnConfig.editable,
+        }
+      : {}),
+    // Update hidden state:
+    ...(notNullOrUndefined(columnConfig.hidden)
+      ? {
+          isHidden: columnConfig.hidden,
+        }
+      : {}),
+    // Add column type metadata:
+    ...(notNullOrUndefined(columnConfig.metadata)
+      ? {
+          columnTypeMetadata: columnConfig.metadata,
+        }
+      : {}),
+    // Add column alignment:
+    ...(notNullOrUndefined(columnConfig.alignment)
+      ? {
+          contentAlignment: columnConfig.alignment,
+        }
+      : {}),
```

Comment:
lodash's `pickBy` might be a nice convenience here (without the necessary comments) Though I guess we do adjust the naming :-(.

```javascript
{
    ...columnProps,
    ...pickBy(columnConfig, notNullOrUndefined)
}
```

Reply:
thanks for the tip 👍  I now have cleaned that up using lodash but decided to go with `merge`. Even if I unify the names between the user-define column config and the column props right now, it might need to diverge in the future.  But `merge` actually ignores undefined values, and thereby lets me write this a lot cleaner. 

```
  return merge(
    { ...columnProps },
    {
      title: columnConfig.title,
      width: columnConfig.width,
      customType: columnConfig.type?.toLowerCase().trim(),
      isEditable: columnConfig.editable,
      isHidden: columnConfig.hidden,
      columnTypeMetadata: columnConfig.metadata,
      contentAlignment: columnConfig.alignment,
    }
  ) as BaseColumnProps
```